### PR TITLE
Release 5.0-beta

### PR DESCRIPTION
**Pull request by LinusDietz**
:tada:

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -41,7 +41,7 @@ apply plugin: LocalizationPlugin
 apply from: 'eclipse.gradle'
 
 group = "org.jabref"
-version = "5.0.0"
+version = "5.0.1"
```
Comment (tobiasdiez) [inline build.gradle]: I think, `5.0.0.1` is sufficient (it just needs to be bigger than `5.0.0.0` for the installer to upgrade automatically).

**Comment by koppor** [comment]
Please remove `-Pdev=true` from the gradle build - or add another workflow, which builds on the branch `release` only. (I had the latter in mind to have a reusable branch)

**Comment by koppor** [comment]
We should finish https://github.com/JabRef/jabref/pull/5676 first to include some fixes in the database handling code, too.

File: build.gradle (hunk 1 above)
Comment (LinusDietz) [reply build.gradle]: needs to be a three digit number. :/

https://docs.microsoft.com/de-de/windows/win32/msi/productversion?redirectedfrom=MSDNhttps://docs.microsoft.com/de-de/windows/win32/msi/productversion?redirectedfrom=MSDN

**Comment by tobiasdiez** [comment]
@koppor I'm ok with merging #5676 asap. We can improve the build script / github actions also later after the release of 5.0 beta.

**Comment by LinusDietz** [comment]
I was wondering what benefits the end users actually have from #5676?

**Comment by koppor** [comment]
Maybe, the `PDev=true` thing can be "fixed" by an environment variable somehow? Maybe not.

**Comment by koppor** [comment]
@LinusDietz The new function `insertEntries` is used when multiple entries are updated remotly in a shared db.

**Comment by LinusDietz** [comment]
> Maybe, the `PDev=true` thing can be "fixed" by an environment variable somehow? Maybe not.

yep, working on it.

**Comment by tobiasdiez** [comment]
I think https://github.com/JabRef/jabref/blob/ea5e632e4e1819673a01bca255304e83924b3dd0/build.gradle#L755 needs to be changed to include `beta` / `alpha` etc

**Comment by koppor** [comment]
project.version should be 5.0-beta - shouldn't it? Like it was at the alpha version

File: build.gradle (hunk 1 above)
Comment (koppor) [reply build.gradle]: I did not notice that we have a new thing here. In 5.0 alpha, we could just put `5.0-alpha` here. See 0c342bdc74a2cea5ebb768e370c121829340ca8e. Back in the install4j-times, we just had a different version number `threeDotVersion`, which we handled separatetly. The enabled us to cope with a) custom JabRef versioning (which is not semantic versioning) and b) versioning required by external tooling.

**Comment by Siedlerchr** [comment]
We just have to specify which version string the installer should pick (it must be numeric)

File: build.gradle (hunk 1 above)
Comment (tobiasdiez) [reply build.gradle]: 5.0 alpha didn't use jpackage. As @LinusDietz said, the WiX toolbox / MSI installer needs a clean three digit number.

**Comment by koppor** [comment]
We should not leave the tag available more than a few minutes in case it is not used. Maybe use the tag "v5.0-beta-test-20191203" or something like that. There might be automatic tooling available checking for releases. In case the tag is changed, it might confuse users. 

See https://help.github.com/en/github/receiving-notifications-about-activity-on-github/watching-and-unwatching-releases-for-a-repository for an example for notifications.

I am ware that with our new setup (https://github.com/JabRef/jabref/pull/5312), the release procedure changed and we need to work on it. - It would be good to have it transparent to the user that we work on the release. Moreover, changing git tags leads to difficulties: https://stackoverflow.com/q/8432953/873282

Thank you for keep working on releasing JabRef 5.0 beta and diving into the hard things of end user-facing software: making a release! 🎉

**Comment by koppor** [comment]
I merged `master`, removed the tag `v5.0-beta` (as I assume this was broken) and added a new tag ` v5.0-beta-test`. If that tag is good, we can creat a tag `v5.0-beta`, can't we?

**Comment by koppor** [comment]
Refs https://github.com/GitTools/GitVersion/issues/1649

**Comment by koppor** [comment]
Maybe releasing only works for the master branch.

**Comment by tobiasdiez** [comment]
It should work now on master with the tag `5.0-beta`. 
The important thing is that
https://github.com/JabRef/jabref/blob/master/GitVersion.yml#L10
needs to be changed directly before a release / tag (because otherwise the build version does not increase) and that
https://github.com/JabRef/jabref/blob/master/GitVersion.yml#L8
is changed accordingly before and after a release (for example before a stable release it should set to be empty, and afterwards it should be set to alpha)

**Comment by koppor** [comment]
I found out we need SemVer in the tag: v5.0.1-beta

Tobias Diez <notifications@github.com> schrieb am Fr., 13. Dez. 2019, 11:56:

**Comment by tobiasdiez** [comment]
Why is this? The last tag on master was `v5.0-alpha` and it worked, right?

**Comment by koppor** [comment]
By using [GitVersion](https://github.com/GitTools/GitVersion), we introduced [SemanticVersioning](https://semver.org/) in JabRef somehow. Our 5.0 alpha was a release. 5.0 beta will be another release. In SemVer, two releases must have different version numbers. So, using `5.0.0` for both 5.0-alpha and 5.0-beta does not make sence to me.

One could argue that our alpha rlease was not a release in the sense of SemVer and that it is perfectly fine with SemVer to use `5.0.0` for both alpha and beta as both are not full releases.

grafik

**Comment by Siedlerchr** [comment]
We could argue that it was pre-release.

**Comment by tobiasdiez** [comment]
We still don't use semantic versioning. The SemVer field produced by GitVersion is not used. We only use AssemblySemVer, which does not use the patch information but instead the number of commits since the last tag (weighted so that it keeps increasing).

To important point is that you can simply take the current installer files and publish them as a release. (If you want add a tag).

**Comment by koppor** [comment]
According to the comment https://github.com/JabRef/jabref/pull/5684#issuecomment-565398044, I created https://github.com/JabRef/jabref/commit/1e9d2478101f69aa845aea7594912fd223ff151a. Hope, this is correct. I am not sure which number to set - as we might release a beta-2.

**Comment by koppor** [comment]
In the chat, I was notified, that I had to revert that commit as no change in that file is necessary any more

**Pull request merged**